### PR TITLE
Added list value getters duckdb_get_list_child and duckdb_get_list_size

### PR DESCRIPTION
**Pull request by prashanthellina**
This adds C-API functions to access data in list values.

```c
idx_t duckdb_get_list_size(duckdb_value value);
duckdb_value duckdb_get_list_child(duckdb_value value, idx_t index);
```

### CODE REVIEW

File: src/main/capi/duckdb_value-c.cpp [hunk 1]
```diff
@@ -329,3 +329,35 @@ duckdb_value duckdb_get_map_value(duckdb_value value, idx_t index) {
...
 	return WrapValue(new duckdb::Value(child_struct[1]));
 }
+
+idx_t duckdb_get_list_size(duckdb_value value) {
+	if (!value) {
+		return 0;
+	}
+
+	auto val = UnwrapValue(value);
+	if (val.type().id() != LogicalTypeId::LIST) {
```
Comment (Tishj) [inline src/main/capi/duckdb_value-c.cpp:339]: Can this also check for LogicalTypeId::MAP, that should also be allowed

Comment (prashanthellina) [reply src/main/capi/duckdb_value-c.cpp]: @Tishj Updated now.

File: src/main/capi/duckdb_value-c.cpp [hunk 2]
```diff
@@ -329,3 +329,35 @@ duckdb_value duckdb_get_map_value(duckdb_value value, idx_t index) {
...
+	return children.size();
+}
+
+duckdb_value duckdb_get_list_child(duckdb_value value, idx_t index) {
+	if (!value) {
+		return nullptr;
+	}
+
+	auto val = UnwrapValue(value);
+	if (val.type().id() != LogicalTypeId::LIST) {
```
[the remark this answers is not in the document]
Comment (prashanthellina) [reply src/main/capi/duckdb_value-c.cpp]: @Tishj Updated now.

**Review by Tishj — APPROVED** [review]
Thanks!

**Comment by Mytherin** [comment]
Thanks! LGTM - could you just resolve the merge conflicts?

**Comment by prashanthellina** [comment]
@Mytherin @Tishj Github says I don't have write access to resolve the conflicts. Is there some other way I can help resolve the conflicts? (Apologies for my weak git foo).

**Comment by Mytherin** [comment]
You need to merge the upstream changes in your fork, e.g.:

```
git pull upstream main
```

**Comment by prashanthellina** [comment]
```bash
(.pyenv) prashanth@prashanth-dev:~/code/duckdb.pr0$ git branch
  feature
* feature-capi-list-value-child
  feature-capi-union-set-tag
  feature-capi-updates
  feature-create-maps-unions-capi
  feature-maps-unions-capi
  feature-table-filters-capi
  main
```

I have these branches in my fork. Initially I had branched off of `feature` into `feature-capi-list-value-child` and then issued the PR.

In this PR activity log above, I see that the "base branch" has been changed from `feature` to `main`.

I did

```bash
git checkout feature-capi-list-value-child
git pull upstream main
```

But I get

```bash
fatal: 'upstream' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

**Comment by prashanthellina** [comment]
@Giorgi Thank you for your response. I realized the same and after some banging around, managed to figure it out!

**Comment by prashanthellina** [comment]
@Mytherin @Tishj The conflicts are resolved and all CI actions passed.

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**